dashboard/app/controllers/reviewable_projects_controller.rb

```diff
@@ -0,0 +1,39 @@
+class ReviewableProjectsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :decrypt_channel_id, only: [:create]
+
+  check_authorization
+  load_and_authorize_resource :reviewable_project, only: [:destroy]
+
+  # POST /reviewable_projects
+  def create
+    authorize! :create, ReviewableProject
+
+    @reviewable_project = ReviewableProject.where(
+      user_id: current_user.id,
+      storage_app_id: @storage_app_id,
+      script_id: params[:script_id],
+      level_id: params[:level_id]
+    ).first_or_create
+
+    if @reviewable_project.save
```

Comment:
I think `first_or_create` will create the new record already (via `.new` or something like that), so I think that you'd want to build the object in the first step, then save it second to decide what the response should be.

Reply:
if you don't want to save above you can use `first_or_initialize`

Reply:
Ah gotcha, thanks 👍